lms/static/js/spec/learner_dashboard/progress_circle_view_spec.js

```diff
@@ -0,0 +1,126 @@
+        var view = null,
+            context = {
+                title: 'XSeries Progress',
+                label: 'Earned Certificates',
+                progress: {
+                    completed: 2,
+                    in_progress: 1,
+                    not_started: 3
+                }
+            },
+            testCircle,
+            testText,
+            initView,
+            getProgress,
+            testProgress;
+
+        testCircle = function(progress) {
+            var $circle = view.$('.progress-circle');
+
+            expect($circle.find('.complete').length).toEqual(progress.completed);
+            expect($circle.find('.incomplete').length).toEqual(progress.in_progress + progress.not_started);
+        };
+
+        testText = function(progress) {
+            var $numbers = view.$('.numbers'),
+                total = progress.completed + progress.in_progress + progress.not_started;
+
+            expect(view.$('.progress-heading').html()).toEqual('XSeries Progress');
+            expect(parseInt($numbers.find('.complete').html(), 10)).toEqual(progress.completed);
+            expect(parseInt($numbers.find('.total').html(), 10)).toEqual(total);
+        };
+
+        getProgress = function(x, y, z) {
+            return {
+                completed: x,
+                in_progress: y,
+                not_started: z
+            };
+        };
+
+        testProgress = function(x, y, z) {
+            var progress = getProgress(x, y, z);
+
+            view = initView(progress);
+            view.render();
+
+            testCircle(progress);
+            testText(progress);
+        };
+
+        initView = function(progress) {
+            var data = $.extend({}, context, {
+                progress: progress
+            });
+
+            return new ProgressCircleView({
+                el: '.js-program-progress',
+                model: new Backbone.Model(data)
+            });
+        };
+
+        beforeEach(function() {
+            setFixtures('<div class="js-program-progress"></div>');
+        });
+
+        afterEach(function() {
+            view.remove();
+        });
+
+        it('should exist', function() {
+            var progress = getProgress(2, 1, 3);
+
+            view = initView(progress);
+            view.render();
+            expect(view).toBeDefined();
+        });
+
+        it('should render the progress circle based on the passed in model', function() {
+            var progress = getProgress(2, 1, 3);
+
+            view = initView(progress);
+            view.render();
+            testCircle(progress);
+        });
+
+        it('should render the progress text based on the passed in model', function() {
+            var progress = getProgress(2, 1, 3);
+
+            view = initView(progress);
+            view.render();
+            testText(progress);
+        });
+
+        it('should render the progress text with only completed courses', function() {
```

Comment:
Nit: you can make all the remaining tests into one with the UI Toolkit's SpecHelpers class:

http://ui-toolkit.edx.org/testing/spec-helpers/